tests/xgboost/test_xgboost_autolog.py

```diff
@@ -295,3 +298,108 @@ def test_xgb_autolog_loads_model_from_artifact(bst_params, dtrain):
 
     loaded_model = mlflow.xgboost.load_model("runs:/{}/model".format(run_id))
     np.testing.assert_array_almost_equal(model.predict(dtrain), loaded_model.predict(dtrain))
+
+
+@pytest.mark.large
+def test_xgb_autolog_gets_input_example(bst_params):
+    mlflow.xgboost.autolog()
+
+    # we cannot use dtrain fixture, as the dataset must be constructed
+    #   after the call to autolog() in order to get the input example
+    iris = datasets.load_iris()
+    X = pd.DataFrame(iris.data[:, :2], columns=iris.feature_names[:2])
+    y = iris.target
+    dataset = xgb.DMatrix(X, y)
+
+    xgb.train(bst_params, dataset)
+    run = get_latest_run()
+
+    model_path = os.path.join(run.info.artifact_uri, "model")
+    model_conf = Model.load(os.path.join(model_path, "MLmodel"))
+
+    input_example = _read_example(model_conf, model_path)
+
+    assert input_example.equals(X[:5])
+
+    pyfunc_model = mlflow.pyfunc.load_model(os.path.join(run.info.artifact_uri, "model"))
+
+    # make sure reloading the input_example and predicting on it does not error
+    pyfunc_model.predict(input_example)
+
+
+@pytest.mark.large
+def test_xgb_autolog_infers_model_signature_correctly(bst_params):
+    mlflow.xgboost.autolog()
+
+    # we cannot use dtrain fixture, as the dataset must be constructed
+    #   after the call to autolog() in order to get the input example
+    iris = datasets.load_iris()
+    X = pd.DataFrame(iris.data[:, :2], columns=iris.feature_names[:2])
+    y = iris.target
+    dataset = xgb.DMatrix(X, y)
+
+    xgb.train(bst_params, dataset)
+    run = get_latest_run()
+    run_id = run.info.run_id
+    artifacts_dir = run.info.artifact_uri.replace("file://", "")
+    client = mlflow.tracking.MlflowClient()
+    artifacts = [x.path for x in client.list_artifacts(run_id, "model")]
+
+    ml_model_filename = "MLmodel"
+    assert str(os.path.join("model", ml_model_filename)) in artifacts
```

Comment:
```suggestion
    assert os.path.join("model", ml_model_filename) in artifacts
```

nit

Reply:
ah, i think this is conflicting with the other XGBoost bugs PR. I will resolve the merges and this should be fixed